superset-frontend/src/dashboard/components/Dashboard.jsx

```diff
@@ -153,16 +151,30 @@ class Dashboard extends React.PureComponent {
     }
 
     if (hasUnsavedChanges) {
-      Dashboard.onBeforeUnload(true);
+      window.addEventListener('beforeunload', Dashboard.showUnsavedMessage);
+    } else {
+      window.removeEventListener('beforeunload', Dashboard.showUnsavedMessage);
+    }
+
+    if (this.canStopPendingQueries && isDashboardLoading(charts)) {
+      window.addEventListener('beforeunload', this.onStopPendingQueries);
     } else {
-      Dashboard.onBeforeUnload(false);
+      window.removeEventListener('beforeunload', this.onStopPendingQueries);
     }
   }
 
   componentWillUnmount() {
     window.removeEventListener('visibilitychange', this.onVisibilityChange);
   }
 
+  onStopPendingQueries() {
```

Comment:
I think it's the `beforeunload` event that triggers the `stopPendingQueries` action. So maybe we can just call this method `stopPendingQueries`?

Reply:
ok. fixed method name.